packages/sanity/src/core/config/document/inspector.ts

```diff
@@ -0,0 +1,88 @@
+import {ValidationMarker} from '@sanity/types'
+import {ButtonTone} from '@sanity/ui'
+import {ComponentType} from 'react'
+
+/** @beta */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface DocumentInspectorProps {
+  onClose: () => void
+}
+
+/** @beta */
+export type DocumentInspectorComponent = ComponentType<DocumentInspectorProps>
+
+/** @beta */
+export interface DocumentInspectorCallbackContext {
+  validation: ValidationMarker[]
+}
+
+/** @beta */
+export interface DocumentInspectorMenuItem {
+  hidden?: boolean | ((props: DocumentInspectorCallbackContext) => boolean)
+  hotkeys?: string[]
+  icon?: ComponentType
```

Comment:
I think it would make sense if `icon` was not optional. Without an icon, the inspector button won't be visible, and without a TS error, it can be difficult to figure out why the button isn't visible. Alternatively, we could have a fallback icon?

<img width="617" alt="Screenshot 2023-05-26 at 14 12 50" src="https://github.com/sanity-io/sanity/assets/15094168/7c006823-916d-403e-8845-ac58e69f6286">